codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithyrust/generator/RustLibraryShimGenerator.java

```diff
@@ -0,0 +1,716 @@
+    return TokenTree.of(
+      evalTemplate(
+        """
+        #[allow(dead_code)]
+        pub fn from_dafny(
+            dafny_value: ::std::rc::Rc<
+                crate::r#$dafnyTypesModuleName:L::$structureName:L,
+            >,
+        ) -> crate::operation::$snakeCaseOperationName:L::$structureName:L {
+            crate::operation::$snakeCaseOperationName:L::$structureName:L::builder()
+                $fluentMemberSetters:L
+                .build()
+                .unwrap()
+        }
+        """,
+        variables
+      )
+    );
+  }
+
+  private RustFile wrappedModule() {
+    final String content = IOUtils.evalTemplate(
+      getClass(),
+      "runtimes/rust/wrapped.rs",
+      MapUtils.merge(serviceVariables(), dafnyModuleVariables())
+    );
+    return new RustFile(Path.of("src", "wrapped.rs"), TokenTree.of(content));
+  }
+
+  private RustFile wrappedClientModule() {
+    final Map<String, String> variables = MapUtils.merge(
+      serviceVariables(),
+      dafnyModuleVariables()
+    );
+    variables.put(
+      "operationImpls",
+      serviceOperationShapes()
+        .map(this::wrappedClientOperationImpl)
+        .collect(Collectors.joining("\n\n"))
+    );
+    final String content = IOUtils.evalTemplate(
+      getClass(),
+      "runtimes/rust/wrapped/client.rs",
+      variables
+    );
+    return new RustFile(
+      Path.of("src", "wrapped", "client.rs"),
+      TokenTree.of(content)
+    );
+  }
+
+  private String wrappedClientOperationImpl(
+    final OperationShape operationShape
+  ) {
+    final Map<String, String> variables = MapUtils.merge(
+      serviceVariables(),
+      dafnyModuleVariables(),
+      operationVariables(operationShape)
+    );
+    return IOUtils.evalTemplate(
+      getClass(),
+      "runtimes/rust/wrapped/client_operation_impl.part.rs",
+      variables
+    );
+  }
+
+  private Path operationsModuleFilePath() {
+    return Path.of("src", "operation");
+  }
+
+  private Path operationModuleFilePath(final OperationShape operationShape) {
+    return operationsModuleFilePath()
+      .resolve(toSnakeCase(operationName(operationShape)));
+  }
+
+  private LocalServiceTrait localServiceTrait() {
+    return service.expectTrait(LocalServiceTrait.class);
+  }
+
+  @Override
+  protected HashMap<String, String> serviceVariables() {
+    final HashMap<String, String> variables = super.serviceVariables();
+
+    final LocalServiceTrait localServiceTrait = localServiceTrait();
+    final String sdkId = localServiceTrait.getSdkId();
+    final String configName = localServiceTrait.getConfigId().getName(service);
+    variables.put("sdkId", sdkId);
+    variables.put("configName", configName);
+    variables.put("snakeCaseConfigName", toSnakeCase(configName));
+
+    return variables;
+  }
+
+  @Override
+  protected String syntheticOperationInputName(OperationShape operationShape) {
+    return operationName(operationShape) + "Input";
```

Comment:
This is fine for now, but we should revisit what our actual preference is here for local services (i.e. this, or Request/Response instead, or use the actual structure name name instead, or...). Maybe cut an issue?